stream-chat-android-client/src/test/java/io/getstream/chat/android/client/socket/ChatSocketServiceImplTest.kt

```diff
@@ -0,0 +1,68 @@
+package io.getstream.chat.android.client.socket
+
+import com.nhaarman.mockitokotlin2.any
+import com.nhaarman.mockitokotlin2.doAnswer
+import com.nhaarman.mockitokotlin2.doReturn
+import com.nhaarman.mockitokotlin2.mock
+import io.getstream.chat.android.client.network.NetworkStateProvider
+import io.getstream.chat.android.client.token.TokenManager
+import org.amshove.kluent.When
+import org.amshove.kluent.calling
+import org.amshove.kluent.shouldBeEqualTo
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+
+internal class ChatSocketServiceImplTest {
+    private lateinit var tokenManager: TokenManager
+    private lateinit var socketFactory: SocketFactory
+    private lateinit var eventsParser: EventsParser
+    private lateinit var networkStateProvider: NetworkStateProvider
+    private lateinit var socketListener: SocketListener
+    private lateinit var socketService: ChatSocketServiceImpl
+
+    @BeforeEach
+    fun setup() {
+        tokenManager = mock()
+        socketFactory = mock()
+        eventsParser = mock()
+        networkStateProvider = mock()
+        socketListener = mock()
+        socketService = ChatSocketServiceImpl.create(
+            tokenManager,
+            socketFactory,
+            eventsParser,
+            networkStateProvider
+        )
+        socketService.addListener(socketListener)
+    }
+
+    @Test
+    fun `Should start connecting to socket when connecting and network connectivity exists`() {
```

Comment:
My personal recommendation. You can read about test namings [here](https://martinfowler.com/bliki/GivenWhenThen.html)  (Martin Fowler's blog).
Usually, if I want to use such naming and there is some condition(s) for the scenario I prefer the following scheme:
`Given arrange When act (Then) should assert`. For your case it is:
``` kotlin
@Test
    fun `Given network connectivity exists When connect Should move state to Connecting`()
```

Reply:
I like to follow a shorter `Should` naming but I think we should discuss this topic and try to standardize our tests naming